instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/engine.py

```diff
@@ -118,23 +119,11 @@ def __init__(
         self._register_event_listener(engine, "checkin", self._pool_checkin)
         self._register_event_listener(engine, "checkout", self._pool_checkout)
 
-    def _get_connection_string(self):
-        drivername = self.engine.url.drivername or ""
-        host = self.engine.url.host or ""
-        port = self.engine.url.port or ""
-        database = self.engine.url.database or ""
-        return f"{drivername}://{host}:{port}/{database}"
-
-    def _get_pool_name(self):
-        if self.engine.pool.logging_name is not None:
-            return self.engine.pool.logging_name
-        return self._get_connection_string()
-
     def _add_idle_to_connection_usage(self, value):
         self.connections_usage.add(
             value,
             attributes={
-                "pool.name": self._get_pool_name(),
+                **self._engine_attrs,
```

Comment:
Do you mean why not just do this?

```
def __init__(self, ...):
    self.engine = weakref.ref(engine)
    ...
```
and then dereference each time the reference is used? Maybe I'm overthinking but my thinking is that doing this way we would risk race conditions with the garbage collector in case of an unexpected crash. For example, `_get_connection_string` and `_get_pool_name` would potentially have up to 5 sequential derefences.

Do you see it differently?